tools/run_defacing_script.pl

```diff
@@ -0,0 +1,687 @@
+
+
+
+=pod
+
+=head3 grep_FileIDs_to_deface($session_id_arr, $modalities_to_deface_arr)
+
+Queries the database for the list of acquisitions' FileID to be used to run the
+defacing algorithm based on the provided list of SessionID and Scan_type to
+restrict the search.
+
+INPUTS:
+  - $session_id_arr          : array of SessionIDs to use when grepping FileIDs
+  - $modalities_to_deface_arr: array of Scan_type to use when grepping FileIDs
+
+RETURNS: hash of matching FileIDs to be used to run the defacing algorithm
+         organized in a hash as follows:
+            {0123}                      # sessionID key
+              {flair}                   # flair scan type key
+                {$FileID} = $File_path  # key = FileID; value = MINC file path
+              {t1}                      # t1 scan type key
+                {$FileID} = $File_path  # key = FileID 1; value = MINC file 1 path
+                {$FileID} = $File_path  # key = FileID 2; value = MINC file 2 path
+
+=cut
+
+sub grep_FileIDs_to_deface {
+    my ($session_id_arr, $modalities_to_deface_arr) = @_;
+
+    # separate the special modalities specified in %SPECIAL_ACQUISITIONS from the
+    # standard scan types
+    my @special_scan_types = keys %SPECIAL_ACQUISITIONS_FILTER;
+    my @special_cases;
+    foreach my $special (@special_scan_types) {
+        # push the special modalities to a new array @special_cases
+        push @special_cases, grep(/$special/, @$modalities_to_deface_arr);
+        # remove the special modalities from the modalities array as they will be
+        # dealt with differently than standard modalities
+        @$modalities_to_deface_arr = grep(! /$special/, @$modalities_to_deface_arr);
+    }
+
+    # base query
+    my $query = "SELECT  SessionID, FileID, Scan_type, File "
+                . "FROM  files f "
+                . "JOIN  mri_scan_type mst ON (f.AcquisitionProtocolID = mst.ID) "
+                . "JOIN  parameter_file pf USING (FileID) "
+                . "JOIN  parameter_type pt USING (ParameterTypeID) "
+                . "WHERE pt.Name='acquisition:image_type' ";
+
+    # add where close for the different standard scan types to deface
+    my @where = map { "mst.Scan_type = ?" } @$modalities_to_deface_arr;
+    $query   .= sprintf(" AND ( %s ", join(" OR ", @where));
+
+    # add where close for the different non-standard scan types to deface
+    # where we will restrict the images to be defaced to the ones with a face base
+    # on parameter_type 'acquisition:image_type'
+    @where  = map { "(mst.Scan_type = ? AND pf.Value LIKE ?)" } @special_cases;
+    $query .= sprintf(" OR %s ) ", join(" OR ", @where));
+
+    # add where close for the session IDs specified to the script if -sessionIDs was set
+    if ($session_id_arr) {
+        @where  = map { "f.SessionID = ?" } @$session_id_arr;
+        $query .= sprintf(" AND (%s) ", join(" OR ", @where));
+    }
+
+    my $sth = $dbh->prepare($query);
+
+    # bind parameters
```

Comment:
I think the entire "bind and execute" block could be replaced by:

```perl
$sth->execute( 
    @$modalities_to_deface_arr, @special_cases, @$session_id_arr
);
```

Reply:
actually, after testing I realized that your suggestion does not work but I still simplified the binding part.